components/user_list_row/user_list_row.tsx

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode} from 'react';
+
+import {ConnectedComponent} from 'react-redux';
+
+import {Client4} from 'mattermost-redux/client';
+
+import ProfilePicture from 'components/profile_picture';
+import UserProfile from 'components/user_profile';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import Nbsp from 'components/html_entities/nbsp';
+import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
+
+import {UserProfile as UserProfileType} from '@mattermost/types/users';
+import {Channel, ChannelMembership} from '@mattermost/types/channels';
+import {TeamMembership} from '@mattermost/types/teams';
+
+import {createSafeId, displayFullAndNicknameForUser} from 'utils/utils';
+
+type Props = {
+    user: UserProfileType;
+    status?: string;
+    extraInfo?: Array<string | JSX.Element>;
+    actions?: Array<ConnectedComponent<any, any>>;
+    actionProps?: {
+        mfaEnabled: boolean;
+        enableUserAccessTokens: boolean;
+        experimentalEnableAuthenticationTransfer: boolean;
+        doPasswordReset: (user: UserProfileType) => void;
+        doEmailReset: (user: UserProfileType) => void;
+        doManageTeams: (user: UserProfileType) => void;
+        doManageRoles: (user: UserProfileType) => void;
+        doManageTokens: (user: UserProfileType) => void;
+        isDisabled?: boolean;
+    };
+    actionUserProps?: {
+        [userId: string]: {
+            channel: Channel;
+            teamMember: TeamMembership;
+            channelMember: ChannelMembership;
+        };
+    };
+    index?: number;
+    totalUsers?: number;
+    userCount?: number;
+};
+
+const UserListRow = ({user, status, extraInfo = [], actions = [], actionProps, actionUserProps = {}, index, totalUsers, userCount}: Props) => {
+    let buttons = null;
+    if (actions) {
+        buttons = actions.map((Action, actionIndex) => {
+            return (
+                <Action
+                    key={actionIndex.toString()}
+                    user={user}
+                    index={index}
+                    totalUsers={totalUsers}
+                    {...actionProps}
+                    {...actionUserProps}
+                />
+            );
+        });
+    }
+
+    // QUICK HACK, NEEDS A PROP FOR TOGGLING STATUS
```

Comment:
Nothing to change here, but I do love seeing 5 year old "quick hack" comments keep getting moved through to new versions of a piece of code 😛